scripts/dss_v2_apdapter.py

```diff
@@ -395,18 +393,12 @@ def _setup(self):
         manifest_entry: MutableJSON
         for file_name, manifest_entry in self.manifest_entries.items():
             file_uuid = manifest_entry['uuid']
-            if file_name == 'project_0.json':
-                self.schema_types[file_uuid] = 'project'
-                self.project_uuid = manifest_entry['uuid']
-            elif file_name == 'links.json':
-                self.schema_types[file_uuid] = 'links'
-            elif manifest_entry['indexed']:  # Metadata files
-                # FIXME: get schema type from 'describeBy' https://github.com/databiosphere/azul/issues/1831
-                match = self.file_name_re.match(file_name)
-                if match:
-                    self.schema_types[file_uuid] = match.group(1)
-                else:
-                    raise Exception(f'Indexed file {file_name} has unknown file name format.')
+            if manifest_entry['indexed']:  # Metadata files
+                described_by = self.indexed_files[file_name]['describedBy']
+                _, _, schema_type = described_by.rpartition('/')
+                self.schema_types[file_uuid] = schema_type
+                if schema_type == 'project':
+                    self.project_uuid = manifest_entry['uuid']
```

Comment:
If there is a way to get the project UUID out of the `provenance.document_id` metadata property that would be more natural but I would also assert that it is `== manifest_entry['uuid']`